### DISCUSSION TITLE
Chatwoot docker local build error webpack

### DISCUSSION THREAD

**Discussion by MuriloFuza** [chatwoot/chatwoot#9992] [Self hosted] [answered]
When following the tutorial on this site ( https://www.chatwoot.com/docs/contributing-guide/project-setup/#docker-for-development ) I encounter the following error in my webpack.

Container log

yarn install v1.22.19
(node:9) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
[1/5] Validating package.json...
error @chatwoot/chatwoot@3.12.0: The engine "node" is incompatible with this module. Expected version "20.x". Got "21.7.2"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
 *  Terminal will be reused by tasks, press any key to close it.

**Comment by facundoRecabarren** [comment]
I am having the same problem!
Did you find a solution?

**Reply by MuriloFuza** [reply, marked as answer]
In this file:
https://github.com/chatwoot/chatwoot/blob/develop/docker/Dockerfile

On line 106 it looks like this
``` SHELL
RUN if [ "$RAILS_ENV" != "production" ]; then \
apk add --no-cache nodejs-current yarn; \
fi
```

You remove nodejs-current and it just works. You can keep the rest, just remove "nodejs-current"

**Reply by facundoRecabarren** [reply]
Thanks, it did the job.
Then I have some other errors but for this it's solved.
I did it another way too:
In package.json `engines` I removed `"node": "20.x"`:
`"engines": {
    "node": "20.x",
    ...
`
